File: src/main/java/com/github/windchopper/tools/eq2/lp/log/DateTimeBuilder.java
package com.github.windchopper.tools.eq2.lp.log;

import com.github.windchopper.tools.eq2.lp.Globals;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped @PartBuilder public class DateTimeBuilder implements LogRecordPartBuilder<DateTime> {

    @Override public String regularExpression() {
        return "^\\((?<dateTimeSeconds>\\d{10})\\)\\[(?<dateTimeFormattedText>)\\]";
    }

    @Override public String toString() {
        return Globals.I18n.bundle.getString("name.wind.tools.eq2.lp.logRecordPartBuilder.dateTime.description");
    }

}

File: src/main/java/com/github/windchopper/tools/eq/lp/log/DateTimeBuilder.kt
package com.github.windchopper.tools.eq.lp.log

import com.github.windchopper.tools.eq.lp.Application
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped @PartBuilder class DateTimeBuilder: LogRecordPartBuilder<DateTime?> {

    override fun regularExpression(): String {
        return "^\\((?<dateTimeSeconds>\\d{10})\\)\\[(?<dateTimeFormattedText>)\\]"
    }

    override fun toString(): String {
        return Application.messages["name.wind.tools.eq2.lp.logRecordPartBuilder.dateTime.description"]!!
    }

}